From 75c9f304808cdb1138a9d5fcd4ceca35fc367e78 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Thu, 18 Sep 2014 14:14:05 -0400
Subject: [scsi] megaraid_sas_fusion: Return correct error value in megasas_get_ld_map_info()

Message-id: <1411049668-28841-6-git-send-email-thenzl@redhat.com>
Patchwork-id: 96156
O-Subject: [RHEL7.1 e-stor PATCH 05/28] megaraid_sas_fusion: Return correct error value in megasas_get_ld_map_info()
Bugzilla: 1088523
RH-Acked-by: Ewan Milne <emilne@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

When no HBA is found we should be returning '-ENXIO' to be consistent
with the other return values.

Signed-off-by: Hannes Reinecke <hare@suse.de>
Acked-by: Kashyap Desai <kashyap.desai@lsi.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 2f8bdfa84a0b814ef66701be1e5dbc88433676e1)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/megaraid/megaraid_sas_fusion.c b/drivers/scsi/megaraid/megaraid_sas_fusion.c
index d10cdf5..9c4b342 100644
--- a/drivers/scsi/megaraid/megaraid_sas_fusion.c
+++ b/drivers/scsi/megaraid/megaraid_sas_fusion.c
@@ -726,7 +726,7 @@ megasas_get_ld_map_info(struct megasas_instance *instance)
 
  if (!fusion) {
   megasas_return_cmd(instance, cmd);
-  return 1;
+  return -ENXIO;
  }
 
  dcmd = &cmd->frame->dcmd;
-- 
1.7.1